Add security threat submission beta api docs (#16242)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/security-threatsubmission.md

---
title: "threatSubmission resource type"

> description: "threat submission base type"

Reviewer: Lines 3 and 16: Please add a description.

> description: "Represents threat submission which is used to submit suspected email, url, file threats to Microsoft 365 Defender."

Reviewer 2: ```suggestion
description: "Represents threat submission which is used to submit suspected email, url, file threats to Microsoft Defender."
```

author: "caigen"
ms.localizationpriority: medium
ms.prod: "security"
doc_type: resourcePageType
---

# threatSubmission resource type

Namespace: microsoft.graph.security

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> It is used to submit suspected spam emails, malware emails, phish emails, malware urls, phish urls, phish email attachments and malware email attachments to Microsoft 365 Defender for Office (MDO), as well as to submit suspicious files to Microsoft 365 Defender for Endpoint (MDE).

Reviewer 3: Please revise this description. What does the article "It" refer to? Generally, a description of a resource starts with a third-person verb, i.e. "Represents" or "Contains". Note that URL is an acronym and should be capitalized. Also, please verify the use of the acronyms; per branding guidelines, we don't form acronyms for trademarked product names like Microsoft.

> It is used to submit suspected spam emails, malware emails, phish emails, malware urls, phish urls, phish email attachments and malware email attachments to Microsoft 365 Defender for Office (MDO), as well as to submit suspicious files to Microsoft 365 Defender for Endpoint (MDE).

Reviewer 2: ```suggestion
Represents threat submission, which is an abstract for representing suspected spam, malware, phish and legitimate emails; malware, phish and legitimate URLs; phish, malware and legitimate email attachments to Microsoft Defender for Office 365(MDO), as well as suspicious files to Microsoft Defender for Endpoint 365(MDE).
```

It could also be used to submit false positive cases that should not be blocked by MDO. For example, not junk emails, safe urls and safe email attachments.

This is an abstract type. Inherits from [entity](../resources/entity.md). Base type of [emailThreatSubmission](../resources/security-emailthreatsubmission.md), [urlThreatSubmission](../resources/security-urlthreatsubmission.md), [fileThreatSubmissin](../resources/security-filethreatsubmission.md).

## Properties
| Property        | Type                       | Description                                                                      |
|:----------------|:---------------------------|:---------------------------------------------------------------------------------|
| adminReview     | [security.submissionAdminReview](../resources/security-submissionadminreview.md)| Specifies the admin review property which constitutes of who reviewed the user submission, when and what was it identified as. |

> | category        | submissionCategory         | Specifies the category of the submission. The possible values are: `notJunk`, `spam`, `phishing` and `malware`.|
> | clientSource    | submissionClientSource     | Specifies the source of the submission. The possible values are: `microsoft` and `other`. |
> | contentType     | submissionContentType      | Specifies the type of content being submitted. The possible values are: `email`, `url`, `file` and `app`.  |
> | createdBy       | [security.submissionUserIdentity](../resources/security-submissionuseridentity.md)     | Specifies who submitted the email as a threat.|
> | createdDateTime | DateTimeOffset             | Specifies when the threat submission was created.                                          |
> | id              | String                     | Specifies the id of threat submission. |
> | result          | [security.submissionResult](../resources/security-submissionresult.md)          | Specifies the result of the analysis performed by Microsoft.  |
> | source          | submissionSource           | Specifies the role of the submitter. The possible values are: `administrator` and `user`.  |
> | status          | longRunningOperationStatus | Indicates whether the threat submission has been analyzed by Microsoft. The possible values are: `notStarted`, `running`, `succeeded`, `failed` and `skipped`. |

Author: ```suggestion
| category        | submissionCategory         | Specifies the category of the submission. The possible values are: `notJunk`, `spam`, `phishing`, `malware` and `unkownFutureValue`.|
| clientSource    | submissionClientSource     | Specifies the source of the submission. The possible values are: `microsoft`,  `other` and `unkownFutureValue`. |
| contentType     | submissionContentType      | Specifies the type of content being submitted. The possible values are: `email`, `url`, `file`, `app` and `unkownFutureValue`.  |
| createdBy       | [security.submissionUserIdentity](../resources/security-submissionuseridentity.md)     | Specifies who submitted the email as a threat.|
| createdDateTime | DateTimeOffset             | Specifies when the threat submission was created.                                          |
| id              | String                     | Specifies the id of threat submission. |
| result          | [security.submissionResult](../resources/security-submissionresult.md)          | Specifies the result of the analysis performed by Microsoft.  |
| source          | submissionSource           | Specifies the role of the submitter. The possible values are: `administrator`,  `user` and `unkownFutureValue`.  |
| status          | longRunningOperationStatus | Indicates whether the threat submission has been analyzed by Microsoft. The possible values are: `notStarted`, `running`, `succeeded`, `failed`, `skipped` and `unkownFutureValue`. |
```

> | adminReview     | [security.submissionAdminReview](../resources/security-submissionadminreview.md)| Specifies the admin review property which constitutes of who reviewed the user submission, when and what was it identified as. |
> | category        | submissionCategory         | Specifies the category of the submission. The possible values are: `notJunk`, `spam`, `phishing`, `malware` and `unkownFutureValue`.|
> | clientSource    | submissionClientSource     | Specifies the source of the submission. The possible values are: `microsoft`,  `other` and `unkownFutureValue`. |
> | contentType     | submissionContentType      | Specifies the type of content being submitted. The possible values are: `email`, `url`, `file`, `app` and `unkownFutureValue`.  |
> | createdBy       | [security.submissionUserIdentity](../resources/security-submissionuseridentity.md)     | Specifies who submitted the email as a threat.|
> | createdDateTime | DateTimeOffset             | Specifies when the threat submission was created.                                          |
> | id              | String                     | Specifies the id of threat submission. |
> | result          | [security.submissionResult](../resources/security-submissionresult.md)          | Specifies the result of the analysis performed by Microsoft.  |
> | source          | submissionSource           | Specifies the role of the submitter. The possible values are: `administrator`,  `user` and `unkownFutureValue`.  |
> | status          | longRunningOperationStatus | Indicates whether the threat submission has been analyzed by Microsoft. The possible values are: `notStarted`, `running`, `succeeded`, `failed`, `skipped` and `unkownFutureValue`. |

Author: ```suggestion
| adminReview     | [security.submissionAdminReview](../resources/security-submissionadminreview.md)| Specifies the admin review property which constitutes of who reviewed the user submission, when and what was it identified as. |
| category        | submissionCategory         | Specifies the category of the submission. Supports `$filter = category eq 'value'`. The possible values are: `notJunk`, `spam`, `phishing`, `malware` and `unkownFutureValue`.|
| clientSource    | submissionClientSource     | Specifies the source of the submission. The possible values are: `microsoft`,  `other` and `unkownFutureValue`. |
| contentType     | submissionContentType      | Specifies the type of content being submitted. The possible values are: `email`, `url`, `file`, `app` and `unkownFutureValue`.  |
| createdBy       | [security.submissionUserIdentity](../resources/security-submissionuseridentity.md)     | Specifies who submitted the email as a threat. Supports `$filter = createdBy/email eq 'value'`. |
| createdDateTime | DateTimeOffset  | Specifies when the threat submission was created. Supports `$filter = createdDateTime ge 2022-01-01T00:00:00Z and createdDateTime lt 2022-01-02T00:00:00Z`.             |
| id              | String                     | Specifies the id of threat submission. |
| result          | [security.submissionResult](../resources/security-submissionresult.md)          | Specifies the result of the analysis performed by Microsoft.  |
| source          | submissionSource           | Specifies the role of the submitter. Supports `$filter = source eq 'value'`. The possible values are: `administrator`,  `user` and `unkownFutureValue`.  |
| status          | longRunningOperationStatus | Indicates whether the threat submission has been analyzed by Microsoft. Supports `$filter = status eq 'value'`. The possible values are: `notStarted`, `running`, `succeeded`, `failed`, `skipped` and `unkownFutureValue`. |
```

| clientSource    | submissionClientSource     | microsoft or other client.                                                       | ❌   | ❌        |

## Relationships
None.

## JSON representation
The following is a JSON representation of the resource.
<!-- {
  "blockType": "resource",
  "keyProperty": "id",
  "@odata.type": "microsoft.graph.security.threatSubmission",
  "baseType": "microsoft.graph.entity",
  "openType": false
}
-->
``` json
{
  "@odata.type": "#microsoft.graph.security.threatSubmission",
  "id": "String (identifier)",
  "tenantId": "String",
  "createdDateTime": "String (timestamp)",
  "contentType": "String",
  "category": "String",
  "source": "String",
  "createdBy": {
    "@odata.type": "microsoft.graph.security.submissionUserIdentity"
  },
  "status": "String",
  "result": {
    "@odata.type": "microsoft.graph.security.submissionResult"
  },
  "adminReview": {
    "@odata.type": "microsoft.graph.security.submissionAdminReview"
  },
  "clientSource": "String"
}
```

> 

Reviewer: Missing: topic for threatSubmissionRoot entity.

---
title: "threatSubmission resource type"
description: "Represent threat submission which is used to submit suspected email, url, file threats to Microsoft 365 Defender."
author: "caigen"
ms.localizationpriority: medium
ms.prod: "security"
doc_type: resourcePageType
---

# threatSubmission resource type

Namespace: microsoft.graph.security

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

This is an abstract type for threat submission. 

> It is used to submit suspected spam emails, malware emails, phish emails, malware urls, phish urls and malware email attachments to Microsoft 365 Defender for Office (MDO), as well as to submit suspicious files to M365 Defender for Endpoint (MDE).

Reviewer 2: ```suggestion
It is used to submit suspected spam emails, malware emails, phish emails, malware urls, phish urls, phish email attachments and malware email attachments to Microsoft 365 Defender for Office (MDO), as well as to submit suspicious files to M365 Defender for Endpoint (MDE).
```

It could also be used to submit false positive cases which should not have been blocked by MDO, e.g, not junk emails, safe urls and safe email attachments.

> Inherits from [entity](../resources/entity.md).

Reviewer: Descriptions in lines 3 and from 16 to 22. 

Let's work together on this to make it clearer for developers. By know, let's add a list of the children types in line 22. 

- Inherits from entity. Base type of {list of children types with reference links for each}.

## Properties
| Property        | Type                       | Description                                                                      |
|:----------------|:---------------------------|:---------------------------------------------------------------------------------|

> | adminReview     | [security.submissionAdminReview](../resources/security-submissionadminreview.md)| User submission admin review |

Reviewer: Line 27, Description: We avoid using 3 or more nouns in a row. In this case "User submission admin review". It is not quite clear what do we mean by this. 

A suggestion would be:
Represents an administrative review of an alert submitted by a user.

> | adminReview     | [security.submissionAdminReview](../resources/security-submissionadminreview.md)| User submission admin review |

Reviewer 2: ```suggestion
| adminReview     | [security.submissionAdminReview](../resources/security-submissionadminreview.md)| It specifies the admin review property which constitutes of who it reviewed the user submission, when and what was it identified as? |
```

> | adminReview     | [security.submissionAdminReview](../resources/security-submissionadminreview.md)| User submission admin review |

Author: ```suggestion
| adminReview     | [security.submissionAdminReview](../resources/security-submissionadminreview.md)| It specifies the admin review property which constitutes of who it reviewed the user submission, when and what was it identified as? |
```

> | category        | submissionCategory         | not junk, spam, phishing or malware.                                             |

Reviewer: Line 28: If this is an enum type, please follow the guidelines for the description here: https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211747/Enumerations?anchor=examples

In this case, a suggestion would be:
The category of the submission. Possible values are: `not junk`, `spam`, `phishing`, `malware` and `unknownFutureValue`.

Reviewer: Please apply this to all the other descriptions of enum types in this table.

Author: Will change but will ignore `unkownFutureValue` since it is a graph suggested common special value.

> | category        | submissionCategory         | not junk, spam, phishing or malware.                                             |

Reviewer 2: ```suggestion
| category        | submissionCategory         | It is used to specify what we are submitting as to Microsoft? The values are are "not junk", "spam", "phishing" or "malware".   |
```

> | clientSource    | submissionClientSource     | microsoft or other client.                                                       |

Reviewer 2: ```suggestion
| clientSource    | submissionClientSource     | It is used to specify what the source for this submission is? Is it Microsoft or other clients which is specified by 'other' |
```

> | adminReview     | [security.submissionAdminReview](../resources/security-submissionadminreview.md)| Represents an administrative review of an threat submitted by a user. |

Author: ```suggestion
| adminReview     | [security.submissionAdminReview](../resources/security-submissionadminreview.md)| It specifies the admin review property which constitutes of who it reviewed the user submission, when and what was it identified as? |
```

> | contentType     | submissionContentType      | email, url, file or app.                                                         |

Reviewer 2: ```suggestion
| contentType     | submissionContentType      | It is used to specify the type of content which is being submitted. The values can be email, URL, file or app.|
```

> | createdBy       | [security.submissionUserIdentity](../resources/security-submissionuseridentity.md)     | The submitter of the threat submission.   |

Reviewer: Line 31, description: Please revise to: The sender of the threat submission.

> | createdBy       | [security.submissionUserIdentity](../resources/security-submissionuseridentity.md)     | The submitter of the threat submission.   |

Reviewer 2: ```suggestion
| createdBy       | [security.submissionUserIdentity](../resources/security-submissionuseridentity.md)     | It is use to specify who submitted the email as a threat.|
```

> | createdDateTime | DateTimeOffset             | When the threat submission was created.                                          |

Reviewer 2: ```suggestion
| createdDateTime | DateTimeOffset             | It is used to specify when the threat submission was created.                                          |
```

> | id              | String                     | Threat submission id. When creation on post, it is not necessary.                |

Reviewer 2: ```suggestion
| id              | String                     | It specifies the id of Threat submission. When creating using post method, it is not required.                |
```

> | result          | [security.submissionResult](../resources/security-submissionresult.md)          | The threat submission check result.  |

Reviewer 2: ```suggestion
| result          | [security.submissionResult](../resources/security-submissionresult.md)          | It is used to specify the result of the analysis which Microsoft had done. |
```

> | source          | submissionSource           | The threat submission source. Administrator or user.                             |

Reviewer 2: ```suggestion
| source          | submissionSource           |It is used to specify who is doing the threat submission? Is it a user or an administrator. |
```

| status          | longRunningOperationStatus | If the threat submission check is finished.                                      |

> | tenantId        | String                     | The tenant id. When creation on post, it is not necessary. Extracted from token. |

Reviewer 2: ```suggestion
| tenantId        | String                     | It denotes the the tenant id from which submission was done. When creation using post method, it is not required. It is extracted from the token of the post API call. |
```

> | category        | submissionCategory         | The category of the submission. The possible values are: `notJunk`, `spam`, `phishing` and `malware`.|
> | clientSource    | submissionClientSource     | The client source of the submission. The possible values are: `microsoft` and `other`. |
> | contentType     | submissionContentType      | The content type of the submission. The possible values are: `email`, `url`, `file` and `app`.  |
> | createdBy       | [security.submissionUserIdentity](../resources/security-submissionuseridentity.md)     | It is use to specify who submitted the email as a threat.|
> | createdDateTime | DateTimeOffset             | It is used to specify when the threat submission was created.                                          |
> | id              | String                     | The id of the threat submission. |
> | result          | [security.submissionResult](../resources/security-submissionresult.md)          | The result of the threat submission.  |
> | source          | submissionSource           | The source of the submission. The possible values are: `administrator` and `user`.  |
> | status          | longRunningOperationStatus | The status of the submission. The possible values are: `notStarted`, `running`, `succeeded`, `failed` and `skipped`. |
> | tenantId        | String                     | It denotes the the tenant id from which submission was done. When creation using post method, it is not required. It is extracted from the token of the post API call. |

Author: ```suggestion
| category        | submissionCategory         | It is used to specify what we are submitting as to Microsoft. The possible values are: `notJunk`, `spam`, `phishing` and `malware`.|
| clientSource    | submissionClientSource     | It is used to specify what the source for this submission is. The possible values are: `microsoft` and `other`. |
| contentType     | submissionContentType      | It is used to specify the type of content which is being submitted. The possible values are: `email`, `url`, `file` and `app`.  |
| createdBy       | [security.submissionUserIdentity](../resources/security-submissionuseridentity.md)     | It is use to specify who submitted the email as a threat.|
| createdDateTime | DateTimeOffset             | It is used to specify when the threat submission was created.                                          |
| id              | String                     | It specifies the id of Threat submission. |
| result          | [security.submissionResult](../resources/security-submissionresult.md)          | It is used to specify the result of the analysis which Microsoft had done.  |
| source          | submissionSource           | It is used to specify who is doing the threat submission? Is it a user or an administrator. The possible values are: `administrator` and `user`.  |
| status          | longRunningOperationStatus | It denotes whether the threat submission has been analyzed by Microsoft or not. The possible values are: `notStarted`, `running`, `succeeded`, `failed` and `skipped`. |
| tenantId        | String                     | It denotes the the tenant id from which submission was done. When creation using post method, it is not required. It is extracted from the token of the post API call. |
```
